### PR TITLE
Fix a few more Flow issues.

### PR DESCRIPTION
**Pull request by jeresig**
Following up from the last PR, a couple more issues that we need to resolve to get it working in Webapp!

### CODE REVIEW

File: packages/wonder-blocks-button/index.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 // @flow
 import Button from "./components/button.js";
+import type {SharedProps} from "./components/button.js";
 
+export type {SharedProps as ButtonProps};
```
Comment (jeresig) [inline packages/wonder-blocks-button/index.js]: A number of places are trying to get our Button SharedProps, which don't exist. I think providing this is a good stopgap.

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit a2fcaa387b238a4c5c2550878374d3ae0704ca96

https://deploy-preview-380--wonder-blocks.netlify.com

File: packages/wonder-blocks-dropdown/components/dropdown.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ import type {StyleType} from "@khanacademy/wonder-blocks-core";
 // NOTE(jeff): Here we share some code for use with PopperJS. Long term,
 // we should either contribute this code to the PopperJS component, or its
 // own non-wonder-blocks package.
-// eslint-disable-next-line import/no-restricted-paths
+// eslint-disable-next-line import/no-restricted-paths $FlowIgnoreMe
```
Comment (jeresig) [inline packages/wonder-blocks-dropdown/components/dropdown.js]: lol, apparently this causes eslint to not work - I'm not sure how to cause both to apply to the next line!

File: packages/wonder-blocks-button/index.js (hunk 1 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-button/index.js]: Interesting... I hadn't considered people using Props types to construct other Props types for other components.  In a future release we should try to be more consistent about exporting those and thinking about how we're naming them.

File: packages/wonder-blocks-button/components/button.js [hunk 3]
```diff
@@ -80,6 +80,11 @@ export type SharedProps = {|
      */
     href?: string,
 
+    /**
+     * A target destination window for a link to open in.
+     */
+    target?: string,
```
Comment (kevinbarabash) [inline packages/wonder-blocks-button/components/button.js]: We can change this to what #378 does later.

Comment (jeresig) [reply packages/wonder-blocks-button/components/button.js]: Yep!

**Pull request merged**